### PR TITLE
feat: support uploading local geo data

### PR DESCRIPTION
**Pull request by arwas11**
Thank you for opening a Pull Request! Before submitting your PR, there are a few things you can do to make sure it goes smoothly:
- [ ] Make sure to open an issue as a [bug/issue](https://github.com/googleapis/python-bigquery-dataframes/issues/new/choose) before writing your code!  That way we can discuss the change, evaluate designs, and agree on the general idea
- [ ] Ensure the tests and linter pass
- [ ] Code coverage does not decrease (if any source code was changed)
- [ ] Appropriate docs were updated (if necessary)

Fixes #308738592 🦕

### CODE REVIEW

**Comment by chelsea-lin** [comment]
The `mypy` tests failed. Could you please help to fix that? `nox -r -s mypy` can help to reproduce tests locally.

File: bigframes/session/__init__.py [hunk 1]
```diff
@@ -703,10 +703,8 @@ def _read_pandas_inline(
         try:
             local_block = blocks.Block.from_local(pandas_dataframe, self)
             inline_df = dataframe.DataFrame(local_block)
-        except pa.ArrowInvalid as e:
-            raise pa.ArrowInvalid(
-                f"Could not convert with a BigQuery type: `{e}`. "
-            ) from e
+        except pa.ArrowInvalid:
+            return None
```
Comment (chelsea-lin) [inline bigframes/session/__init__.py:707]: Please help to add comments here. For example, `An error is thrown for unsupported types, such as geo.`

**Review by chelsea-lin — CHANGES_REQUESTED** [review]

**Review by chelsea-lin — APPROVED** [review]
Thanks! LGTM

**Comment by tswast** [comment]
Merging. `doctest` and Samples test failures appear to be unrelated.

**Pull request merged**